main/src/ca/uwaterloo/flix/language/phase/jvm/GenExpression.scala

```diff
@@ -116,11 +116,7 @@ object GenExpression {
         addSourceLine(mv, loc)
         mv.visitLdcInsn(s)
 
-      case Ast.Constant.Regex(patt) =>
-        addSourceLine(mv, loc)
-        mv.visitLdcInsn(patt.pattern)
-        mv.visitMethodInsn(INVOKESTATIC, JvmName.Regex.toInternalName, "compile",
-          AsmOps.getMethodDescriptor(List(JvmType.String), JvmType.Regex), false)
+      case _ => throw InternalCompilerException(s"Unexpected Constant $cst", loc)
```

Comment:
Can we not remove it from `Ast.Constant`? or is that not possible?

At minimum, I think we should replace `_` by matching on the Regex.

Reply:
> Can we not remove it from `Ast.Constant`? or is that not possible?

Remove what exactly? :)

Reply:
But can we drop `Ast.Constant.Regex` or is that used earlier in the compiler?

Reply:
I see its used earlier.